source/geometries/Next100Shielding.cc

```diff
@@ -86,128 +93,182 @@ namespace nexus {
+                                                    lead_thickness_/2.,
+                                                    pedestal_support_bottom_thickness/2.);
+
+    G4Box* pedestal_support_beam_top = new G4Box("PEDESTAL_SUPPORT_BEAM_TOP",
+                                                 pedestal_x_/2.,
+                                                 pedestal_support_top_thickness/2.,
+                                                 pedestal_support_top_length/2.);
+
+    G4Box* pedestal_beam_lat = new G4Box("PEDESTAL_BEAM_LAT",
+                                         pedestal_x_/2.,
+                                         lead_thickness_/2.,
+                                         pedestal_lateral_beam_thickness/2.);
+
+    G4LogicalVolume* pedestal_support_beam_bottom_logic = new G4LogicalVolume(pedestal_support_beam_bottom,
+                                                                              MaterialsList::Steel316Ti(), "PEDESTAL_SUPPORT_BOTTOM");
+
+    G4LogicalVolume* pedestal_support_beam_top_logic = new G4LogicalVolume(pedestal_support_beam_top,
+                                                                           MaterialsList::Steel316Ti(), "PEDESTAL_SUPPORT_TOP");
+
+    G4LogicalVolume* pedestal_beam_lat_logic = new G4LogicalVolume(pedestal_beam_lat,
+                                                                   MaterialsList::Steel316Ti(), "PEDESTAL_SUPPORT_LAT");
+
+    G4double pedestal_y_pos = -lead_y_/2. + lead_thickness_/2.;
+    G4double pedestal_top_y_pos = -(shield_y_/2. + steel_thickness_/2.) + pedestal_support_top_thickness/2.; //caution: it is refered to air-box
+    new G4PVPlacement(0, G4ThreeVector(0., pedestal_y_pos, support_beam_dist_/2.),
+                      pedestal_support_beam_bottom_logic, "PEDESTAL_SUPPORT_BEAM_BOTTOM_1", lead_box_logic, false, 0);
+
+    new G4PVPlacement(0, G4ThreeVector(0., pedestal_y_pos, -support_beam_dist_/2.),
+                      pedestal_support_beam_bottom_logic, "PEDESTAL_SUPPORT_BEAM_BOTTOM_2", lead_box_logic, false, 0);
 
+    new G4PVPlacement(0, G4ThreeVector(0., pedestal_top_y_pos, support_beam_dist_/2.),
+                      pedestal_support_beam_top_logic, "PEDESTAL_SUPPORT_BEAM_TOP_1", air_box_logic_, false, 0);
 
+    new G4PVPlacement(0, G4ThreeVector(0., pedestal_top_y_pos, -support_beam_dist_/2.),
+                      pedestal_support_beam_top_logic, "PEDESTAL_SUPPORT_BEAM_TOP_2", air_box_logic_, false, 0);
+
+    new G4PVPlacement(0, G4ThreeVector(0., pedestal_y_pos, support_beam_dist_/2. + support_lat_dist_),
+                      pedestal_beam_lat_logic, "PEDESTAL_BEAM_LAT_1", lead_box_logic, false, 0);
+
+    new G4PVPlacement(0, G4ThreeVector(0., pedestal_y_pos, -support_beam_dist_/2. - support_lat_dist_),
+                      pedestal_beam_lat_logic, "PEDESTAL_BEAM_LAT_2", lead_box_logic, false, 0);
 
     // SETTING VISIBILITIES   //////////
     if (visibility_) {
```

Comment:
I would add the visibility behaviour of the new pedestal volumes. We can set the same one as the rest of the structure, for instance, and make them invisible if visibility is zero.